packages/mail-client/src/mail-client.ts

```diff
@@ -0,0 +1,187 @@
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain host or port information as properties. Please configure in the "Additional Properties" of the destination.`
+    );
+  }
+  return { host, port: parseInt(port) };
+}
+
+function getCredentials(
+  destination: Destination,
+  originalProperties: Record<string, any>
+): { username: string; password: string } {
+  const username = originalProperties['mail.user'];
+  const password = originalProperties['mail.password'];
+
+  if (!(username && password)) {
+    throw Error(
+      `The destination '${toDestinationNameUrl(
+        destination
+      )}' does not contain user or password information as properties. The mail-client only supports basic authentication, so "user" and "password" are required.`
+    );
+  }
+  return { username, password };
+}
+
+function createTransport(
+  mailDestination: MailDestination
+): Transporter<SentMessageInfo> {
+  return nodemailer.createTransport({
+    host: mailDestination.host,
+    port: mailDestination.port,
+    // Defines if the connection should use SSL (if true) or not (if false).
+    secure: false,
+    auth: {
+      user: mailDestination.username,
+      pass: mailDestination.password
+    },
+    // TODO: Uploading certificates on CF like HTTP destination is not applicable for MAIL destination.
+    // Provide an API option for the users, so they can pass it as parameter for the nodemailer.
+    // The `tls` is the right key:
+    //     tls: {
+    //       cert: xxx,
+    //       ca: xxx,
+    //       rejectUnauthorized: xxx
+    //     }
+    tls: {
+      /**
+       * If true the server will reject any connection which is not
+       * authorized with the list of supplied CAs. This option only has an
+       * effect if requestCert is true.
+       */
+      /** Disable tls config to fix the self signed certificate error. */
+      rejectUnauthorized: false
+    }
+  });
+}
+
+function buildMailOptionsFromDestination(
+  mailDestination: MailDestination
+): Partial<MailOptions> {
+  const from = mailDestination.from;
+  return { from };
+}
+
+async function sendMailWithNodemailer<T extends MailOptions>(
+  mailDestination: MailDestination,
+  ...mailOptions: T[]
+): Promise<MailResponse[]> {
+  const transporter = createTransport(mailDestination);
+  const mailOptionsFromDestination =
+    buildMailOptionsFromDestination(mailDestination);
+
+  const response: MailResponse[] = [];
+  for (const mailOptionIndex in mailOptions) {
+    logger.debug(
+      `Sending email ${mailOptionIndex + 1}/${mailOptions.length}...`
+    );
+    response[mailOptionIndex] = await transporter.sendMail({
+      ...mailOptionsFromDestination,
+      ...mailOptions[mailOptionIndex]
+    });
+    logger.debug(
+      `...email ${mailOptionIndex + 1}/${mailOptions.length} for subject "${
+        mailOptions[mailOptionIndex].subject
+      }" was sent successfully.`
+    );
+  }
+
+  transporter.close();
```

Comment:
[pp] I would like to call it `transport` as the function also says `createTransport`.
```suggestion
  const transport = createTransport(mailDestination);
  const mailOptionsFromDestination =
    buildMailOptionsFromDestination(mailDestination);

  const response: MailResponse[] = [];
  for (const mailOptionIndex in mailOptions) {
    logger.debug(
      `Sending email ${mailOptionIndex + 1}/${mailOptions.length}...`
    );
    response[mailOptionIndex] = await transport.sendMail({
      ...mailOptionsFromDestination,
      ...mailOptions[mailOptionIndex]
    });
    logger.debug(
      `...email ${mailOptionIndex + 1}/${mailOptions.length} for subject "${
        mailOptions[mailOptionIndex].subject
      }" was sent successfully.`
    );
  }

  transport.close();
```